### DISCUSSION TITLE
Issue with PMD custom ruleset

### DISCUSSION THREAD

**Discussion by Rahul051992** [pmd/pmd#4525] [Q&A]
Hello Team
I have created a custom pmd ruleset for apex in folder-rulesets/apex/
But when i am trying to perform a pmd check on my apex code using commandline i am getting
"Cannot resolve rule/ruleset reference.Make sure the resource is a valid file or URL and is on the CLASSPATH" for my custom ruleset.  
![pmdrulesetissue](https://user-images.githubusercontent.com/16416724/235312464-7b7b2204-4f83-45e7-b4d5-4366edf07e80.JPG)
The image of my custom ruleset xml file-
![MyRulesxml](https://user-images.githubusercontent.com/16416724/235313496-f28380c8-7a05-47a9-a226-e5b0d8df5a07.JPG)

**Comment by oowekyala** [comment]
The error message means that `rulesets/apex/MyRules.xml` cannot be found on the classpath and is also not a valid file path (on windows you should use `\` as path separator). Unless you have bundled your ruleset in a jar that is on the classpath, it will not be picked up on. A simpler method would be not to use a classpath reference but instead a file path. I see you are in `C:\...\Salesforce Resources`. You can just place your `MyRules.xml` in that directory and use `-R MyRules.xml`. If you want to place it elsewhere than the working directory, then fix the path separators.

**Comment by Rahul051992** [comment]
Hello oowekyala
I tried 'file path' solution I am  getting the same error. I would like to try the 'jar-classpath' way. Can you guide me on what all files needs to be included in the jar file i.e. do i need to include all such as 'Documentation.xml','Security.xml' .xml files being referenced in my custom ruleset file. and while running the check should i provide the path of my jar file or my custom ruleset xml file

I am new to this tool and i have never worked with jar files. I bundled my ruleset into a jar file and added a classpath as well,but i am receiving the same error

Thank you
